store/store.go

```diff
@@ -73,6 +74,29 @@ func (bs *BlockStore) LoadBlock(height int64) *types.Block {
 	return block
 }
 
+// LoadBlockByHash returns the block with the given height.
+// If no block is found for that height, it returns nil.
+func (bs *BlockStore) LoadBlockByHash(hash bytes.HexBytes) *types.Block {
+	var blockMeta = bs.LoadBlockMetaByHash(hash)
```

Comment:
I think what we need here is to store `hash => height` relation and use it to retrieve the block

```
bz := bs.db.Get(calcBlockHashKey(hash))
	if len(bz) == 0 {
		return nil
	}
height := parseHeight(string(bz))
return LoadBlock(height)
```